### PR TITLE
mcs: solve stream error when forward tso

### LINKED ISSUE #7320: The error of forwarding TSO should be handled
## Bug Report

https://github.com/tikv/pd/blob/ab8bf7b7a62d43abd9a8d5213fc3b5855472bd9e/server/grpc_service.go#L2626-L2653
https://github.com/tikv/pd/blob/ab8bf7b7a62d43abd9a8d5213fc3b5855472bd9e/server/grpc_service.go#L2635

The error of sending the request is not properly handled. We should do the error handling in case there is something wrong when forwarding the request to get TSO.

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close #7320

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by rleungx — APPROVED** [review]

File: server/grpc_service.go [hunk 1]
```diff
@@ -2622,42 +2617,53 @@ func (s *GrpcServer) getGlobalTSOFromTSOServer(ctx context.Context) (pdpb.Timest
 		forwardStream tsopb.TSO_TsoClient
 		ts            *tsopb.TsoResponse
 		err           error
+		ok            bool
 	)
+	handleStreamError := func(err error) (needRetry bool) {
+		if strings.Contains(err.Error(), errs.NotLeaderErr) {
+			s.tsoPrimaryWatcher.ForceLoad()
+			time.Sleep(retryIntervalRequestTSOServer)
```
Comment (CabinfeverB) [inline server/grpc_service.go:2625]: I prefer to move the interval of retry out of this function

**Review by CabinfeverB — COMMENTED** [review]
rest LGTM

File: server/grpc_service.go [hunk 2]
```diff
@@ -2648,6 +2646,7 @@ func (s *GrpcServer) getGlobalTSO(ctx context.Context) (pdpb.Timestamp, error) {
 		err = forwardStream.Send(request)
 		if err != nil {
 			if needRetry := handleStreamError(err); needRetry {
+				time.Sleep(retryIntervalRequestTSOServer)
```
Comment (CabinfeverB) [inline server/grpc_service.go:2649]: How about move it into the beginning of the `for` statement? Check if i == 0

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**